routes/userapp.js

```diff
@@ -1,9 +1,103 @@
+const https = require('https');
 const express = require('express');
 const router = express.Router();
 
-/* GET users listing. */
-router.get('/', (req, res) => {
-	res.render('userapp');
+const mdb = require('./dbapi');
+const db = new mdb.Database();
+
+const config = require('./config');
+
+/*
+curl -H "Content-Type: application/json" -H "Accept: application/json" \
+-X POST \
+-d '{"name":"user_name"}' http://localhost:3000/login
+OR
+-d '{"auth_token": "..."}' http://localhost:3000/login
+*/
+router.post('/login', function(req, res, next) {
+  // Check if auth_token is provided
+  if (req.body.hasOwnProperty(config.AUTH_TOKEN)) {
+    getProfileFromGoogle(req.body[config.AUTH_TOKEN], (googleName, email) => {
+      // Once email is retrieved, check if already exists in db
+      db.checkUserExists(email, (_id, name) => {
+        if (_id) return loginResponse(_id, name, res);
+        createUser(googleName, email, res);
+      });
+    });
+  } else if (req.body.hasOwnProperty(config.NAME)
+    && req.body[config.NAME].length != 0) {
+    // Create user without email
+    createUser(req.body[config.NAME], undefined, res);
+  } else res.status(400).send('Name or auth_token not provided.');
+});
+  
+let createUser = function(name, email, res) {
+  db.createUser(name, email, (_id, name) => {
+    if (_id) return loginResponse(_id, name, res);
+    res.status(500).send('Error occurred in user creation.');  
+  });
+}
+
+let loginResponse = function(_id, name, res) {
+  res.cookie(config.COOKIE_NAME, _id.toString())
+    .send({'user' : {'name': name}});
+}
+
+let getProfileFromGoogle = function(accessToken, callback) {
+  var options = {
```

Comment:
merging, but see if you can change it to use an NPM module.
That ways whenever these APIs change a simple upgrade of package would keep us green.